mountpoint-s3-client/src/object_client.rs

```diff
@@ -264,6 +272,24 @@ pub enum DeleteObjectError {
     NoSuchBucket,
 }
 
+/// Result of a [`copy_object`](ObjectClient::copy_object) request
+#[derive(Debug)]
+#[non_exhaustive]
+pub struct CopyObjectResult {
+    // TODO: Populate this struct with return fields from the S3 API, e.g., etag.
+}
+
+/// Errors returned by a [`copy_object`](ObjectClient::copy_object) request
+#[derive(Debug, Error, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum CopyObjectError {
```

Comment:
Are these the only errors we can surface? Not even e.g. `NotFound`?

Reply:
for any non existing bucket or key we just get a 404 response status, adding that as a criteria for NotFound error.